### PR TITLE
[TRA] Train predictions fix

### PR DESCRIPTION
**Pull request by emilydinan**
**Patch description**
Issue flagged thanks to #1914. Need to filter out padded candidates from the ordering, otherwise we get an index error.

### CODE REVIEW

**Comment by alexholdenmiller** [comment]
hm this might be related to the diff I landed

**Comment by emilydinan** [comment]
> hm this might be related to the diff I landed

i actually think it was broken before too!

**Comment by alexholdenmiller** [comment]
#1894 changed the similar code in eval_step but not in _get_train_preds

File: parlai/core/torch_ranker_agent.py [hunk 1]
```diff
@@ -277,11 +277,20 @@ def _get_train_preds(self, scores, label_inds, cands, cand_vecs):
...
+            elif cand_vecs.dim() == 3:
+                cand_list = cands[i]
+            if len(ordering) != len(cand_list):
+                # ignore padding
+                for x in ordering:
+                    if x < len(cand_list):
+                        preds.append(cand_list[x])
+                        break
+            else:
+                preds.append(cand_list[ordering[0]])
```
Comment (alexholdenmiller) [inline parlai/core/torch_ranker_agent.py]: return Output

**Comment by stephenroller** [comment]
File an issue for testing a couple TRA models with every variant for -cands?

**Pull request merged**